edit/edit_test.go

```diff
@@ -172,3 +172,43 @@ func TestAddValueToListAttribute(t *testing.T) {
 		}
 	}
 }
+
+func TestUseImplicitName(t *testing.T) {
+	tests := []struct {
+		input            string
+		expectedRuleLine int
+		wantErr          bool
+		wantRootErr      bool
+	}{
+		{`rule()`, 1, false, false},
+		{`rule(name="a")
+		  rule(name="b")
+		  rule()`, 3, false, false},
+		{`rule() rule() rule()`, 1, true, false},
+		{`rule()`, 1, true, true},
+	}
+
+	for _, tst := range tests {
+		path := "foo/BUILD"
+		if tst.wantRootErr {
+			path = "BUILD"
+		}
+		bld, err := build.Parse(path, []byte(tst.input))
+		if err != nil {
+			t.Error(err)
```

Comment:
Because we're in a loop, it's useful to prefix errors with a descriptive string to say which test case this is failing for - the test runtime will only output the line number, which doesn't tell you which thing failed. Accordingly, it's fairly common to have a `description` string in your tests struct, and here do `t.Error(tst.description, err)`

You could also use this below where you're currently using tst.input for the same purpose (but two of your test cases have the same input :))